### PR TITLE
[chore] disheartening switch off of a OSS cpu test

### PR DESCRIPTION
**Pull request by blefaudeux**
# Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
This test runs on CPU, and typically never fails, on agents which are cuda enabled (it does use Gloo and device==cpu). The newer jobs which don't have cuda randomly fail on this very same test with python3.9, even if to the best of my knowledge nothing should change. This test remains active for gpu enabled agents, even if 

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/optim/test_oss.py [hunk 1]
```diff
@@ -564,6 +564,7 @@ def closure():
     dist.destroy_process_group(process_group)
 
 
+@skip_if_py39_no_cuda
 def test_multiple_groups():
```
Comment (blefaudeux) [inline tests/optim/test_oss.py:568]: the failure case is specifically on agents which do not have cuda, even if this should not matter to this specific test (gloo and cpu).

**Comment by blefaudeux** [comment]
see https://app.circleci.com/pipelines/github/facebookresearch/fairscale/1533/workflows/9fbae0c7-2586-4688-86fa-306e1aaf21ce/jobs/7202 (and the rest of the history, one step up) to have an idea of the flakiness of this test

**Comment by min-xu-ai** [comment]
thanks for including the details. it is sad, but sure.

**Review by min-xu-ai — APPROVED** [review]

**Comment by blefaudeux** [comment]
> thanks for including the details. it is sad, but sure.

I think it would be nice to debug eventually, but we're a bit short handed for that, and it's not really on the critical path to mission, so compromising..

**Pull request merged**